client/components/Editor.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { useState } from 'react';
+import 'codemirror/lib/codemirror.css';
+import 'codemirror/theme/material.css';
+import 'codemirror/mode/xml/xml';
+import 'codemirror/mode/javascript/javascript';
+import 'codemirror/mode/css/css';
+import { Controlled as ControlledEditor } from 'react-codemirror2';
+
+export default function Editor(props) {
+  const {
+    language,
+    displayName,
+    value,
+    onChange
+  } = props;
+  const [open, setOpen] = useState(true);
+
+  function handleChange(editor, data, value) {
+    onChange(value);
+  }
+
+  return (
+    <div className={`editor-container ${open ? '' : 'collapsed'}`}>
+      <div className="editor-title">
+        {displayName}
+        <i
+          type="button"
+          className="fa-solid fa-arrow-rotate-left adjust-collapsed"
```

Comment:
You should use two icons for open or closed. I recommend chevrons

https://fontawesome.com/search?p=1&q=chevron&s=solid%2Cbrands

Reply:
chevrons is too complicated, cause for windows I collapsed from left and right, and for mobile I collapsed from up and down, so instead of chevron I used expand-alt and compress-alt